### PR TITLE
[manage] Purge old request specs for deleted instances

### PR DESCRIPTION
**Pull request by grandchild**
Add `nova-manage api_db purge_request_specs` command to clean up db for the upcoming mariadb migration.

### CODE REVIEW

**Comment by grandchild** [comment]
I don't expect this code needing to survive to the next nova version, hence the somewhat lackluster copy-paste-extension in `nova/objects/{instance_mapping,request_spec}.py`.

**Comment by grandchild** [comment]
This is now tested in qa-de-1, and successfully deleted a few rows. Please have a look.

File: nova/cmd/manage.py [hunk 1]
```diff
@@ -812,6 +812,75 @@ def version(self):
...
+        """Removes old request specs for deleted instances"""
+        older_than = int(older_than)
+        max_number = int(max_number)
+
+        admin_context = context.get_admin_context()
+        extant_instance_uuids = [
+            # There's no need to join the default security_groups and
+            # instance_info_caches tables, hence explicit empty
+            # 'columns_to_join'.
+            i.get('uuid', None) for i in db.instance_get_all(
```
Comment (joker-official) [inline nova/cmd/manage.py:836]: Does `get()` not return `None` here by default?

File: nova/cmd/manage.py [hunk 2]
```diff
@@ -812,6 +812,75 @@ def version(self):
...
+                admin_context, columns_to_join=[])
+        ]
+        finished = False
+        while not finished:
+            finished = self._purge_request_specs_and_instance_mappings(
+                admin_context, extant_instance_uuids, dry_run, older_than,
```
Comment (joker-official) [inline nova/cmd/manage.py:842]: Shouldn't we filter out the `None`s out of `extant_instance_uuids` before passing on the list?

Comment (grandchild) [reply nova/cmd/manage.py:839]: i couldn't readily find any instance rows that didn't have a uuid, but i fixed it anyway.

**Pull request merged**